packages/wxt/src/core/builders/vite/plugins/devServerGlobals.ts

```diff
@@ -15,9 +15,9 @@ export function devServerGlobals(
 
       return {
         define: {
-          __DEV_SERVER_PROTOCOL__: JSON.stringify('ws:'),
-          __DEV_SERVER_HOSTNAME__: JSON.stringify(server.hostname),
-          __DEV_SERVER_PORT__: JSON.stringify(server.port),
+          __DEV_SERVER_ORIGIN__: JSON.stringify(
+            server.origin.replace(/^http(s):/, 'ws$1:'),
```

Comment:
The regex replace converts `http:` to `ws:` and `https:` to `wss:`

Reply:
@alecdwm @Sokwva Fixed by https://github.com/wxt-dev/wxt/pull/1707